crates/module-system/module-implementations/sov-evm/src/rpc/mod.rs

```
@@ -360,12 +366,88 @@ where
         let gas_used_with_margins = (total_gas_used * 3) / 2 + RELATIVE_MARGIN; // gas * 1.5 + 100_000
         Ok(U64::from(gas_used_with_margins))
     }
+
+    /// Handler for: `debug_traceTransaction`
+    #[rpc_method(name = "traceTransaction")]
+    pub fn debug_trace_transaction(
+        &self,
+        tx_hash: B256,
+        opts: Option<GethDebugTracingOptions>,
+        state: &mut ApiStateAccessor<S>,
+    ) -> RpcResult<GethTrace> {
+        let opts = opts.unwrap_or_default();
+        let index = self.get_tx_index_by_hash(&tx_hash, state).unwrap();
+        let tx = self.transaction(index, state).unwrap();
+
+        let mut state = state
+            .get_archival_state(RollupHeight::new(tx.block_number - 1))
+            .unwrap();
```

with pruning this can fail. Let's return an Rpc error in that case.